bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/link/ItemChannelLinkConfigDescriptionProvider.java

```diff
@@ -123,8 +125,35 @@ private boolean isSupportedItemType(ProfileType profileType, Item item) {
                 || profileType.getSupportedItemTypes().contains(item.getType());
     }
 
-    private boolean isSupportedChannelType(TriggerProfileType profileType, Channel channel) {
-        return profileType.getSupportedChannelTypeUIDs().isEmpty()
-                || profileType.getSupportedChannelTypeUIDs().contains(channel.getChannelTypeUID());
+    private boolean isSupportedChannelType(ProfileType profileType, Channel channel, @Nullable Locale locale) {
+        @Nullable
+        ChannelKind supportedChannelKind = profileType.getSupportedChannelKind();
+        if (supportedChannelKind != null && supportedChannelKind != channel.getKind())
+            return false;
```

Comment:
```suggestion
        if ((channel.getKind() == STATE && !(profileType instanceof StateProfileTypeImpl)) || (channel.getKind() == TRIGGER && !(profileType instanceof TriggerProfileTypeImpl) {
            return false;
        }
        @Nullable
        ChannelKind supportedChannelKind = profileType.getSupportedChannelKind();
        if (supportedChannelKind != null && supportedChannelKind != channel.getKind())
            return false;
```

You don't need the new method in `ProfileType` then.

Reply:
I did this because I personally find it horrible to resort to `instanceof` to figure out what to do. I mean, the whole point of an interface is to abstract out the implementing class, or even derived interfaces.
If in the future there is another profile type, there would be another interface but there would not be a need to come back to this code because the `getSupportedChannelKind` already provides the answer for this.
Also, if there is a new member in the `ChannelKind` enumeration, once again the code above does not need to be changed, it's the same method that I introduced that would be called.

Reply:
IMO very unlikely, but ok. Let's keep the methods.